Maple2.Model/Game/FishCatch.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Runtime.InteropServices;
+using Maple2.Model.Enum;
+using Maple2.Model.Metadata;
+
+namespace Maple2.Model.Game;
+
+[StructLayout(LayoutKind.Sequential, Pack = 4, Size = 12)]
+public struct FishCatch {
```

Comment:
Generally I only create readonly structs unless there's a good reason otherwise. Currently you are constructing `FishCatch` and then passing it to `AddMastery`. Instead you can pass `FishTable.Entry fishEntry, CaughtFishType type` to `AddMastery` and construct `FishCatch` there. That way you will not need to mutate this struct and can make it readonly.

---

Perhaps you can just eliminate this struct entirely though and pass the 4 values in it as parameters to `IncreaseMastery` since it's not used anywhere else.